src/Microsoft.PowerShell.Commands.Management/Microsoft.PowerShell.Commands.Management.csproj

```diff
@@ -5,7 +5,18 @@
     <NoWarn>$(NoWarn);CS1570</NoWarn>
     <AssemblyName>Microsoft.PowerShell.Commands.Management</AssemblyName>
   </PropertyGroup>
-
+  <PropertyGroup>
+    <OnOSX Condition="'$([System.Runtime.InteropServices.RuntimeInformation]::IsOSPlatform($([System.Runtime.InteropServices.OSPlatform]::OSX)))' == 'true'">true</OnOSX>
```

Comment:
Please don't prevent cross compilation.  Pass in the property from the build.psm1.  You can then default the property if it is not set from the commandline.

Reply:
Example:
https://github.com/PowerShell/PowerShell/blob/c17d5a7a2796cc31b504c65724b4f8a5292ee344/PowerShell.Common.props#L115
https://github.com/PowerShell/PowerShell/blob/c17d5a7a2796cc31b504c65724b4f8a5292ee344/build.psm1#L347

Reply:
Fixed to be a runtime thing.